go/tools/extract_package/extract_test.go

```diff
@@ -1,21 +1,81 @@
 package main
 
 import (
+	"io/ioutil"
+	"os"
 	"testing"
 )
 
 func TestExtract(t *testing.T) {
-	for _, fname := range []string{
-		"extract.go",
-		"extract_test.go",
-		"ignored.go",
+	for _, spec := range []struct {
+		src  string
+		want string
+	}{
+		// These example inputs also illustrate the reason why we
+		// cannot simply replace extract_package with sed(1) or awk(1).
+		{
+			src:  `package main`,
+			want: "main",
+		},
+		{
+			src:  `/* package main */ package example`,
+			want: "example",
+		},
+		{
+			src:  `package /* package another */ example // package yetanother`,
+			want: "example",
+		},
+		{
+			src: `package /*
+			long comments here
+*/			      example`,
+			want: "example",
+		},
+		{
+			src: `// +build exampletag
+
+package example
+`,
+			want: "example",
+		},
 	} {
-		name, err := extract(fname)
-		if err != nil {
-			t.Errorf("extract(%q) fails with %v; want success", fname, err)
+		withTempFile(t, spec.src, func(fname string) {
```

Comment:
this looks a little bit like python (or ruby?).


  t, err :=  ioutil.TempFile()
  if err != nil  { t.Fatal(err) }
  if _, err :=  ioutil.WriteFile(.., 0644); err != nil { t.Fatal(err) }
  ..


you don't have to add so many cases because you only add 3 lines on top of go/parser.

In fact, I'd even consider dropping the test entirely. If you break this, you should have an end2end test that catches it anyway.

Come to think of it: you do have an example, but do you have an e2e test that builds a binary and checks that it works as expected? There used to be one, but not sure what happened after the rules got moved out of bazel.